### ISSUE TITLE
referrals less than 1.00 using comma separator create 0.00 referral amount

### ISSUE THREAD

**Issue by gspice** [AffiliateWP/AffiliateWP#1972]
User settings are:

* currency Euro
* decimal separator comma
* default referral rate type: Percentage

On Woo product, set referral rate to 0,91  for .91 cents.   When the referral is generated it comes at 0.00.  If using default flat referral rate type it correctly shows 0.91 for the referral.  I verified user updated to latest version of AffWP plugin.   

https://secure.helpscout.net/conversation/309390582/51582/?folderId=205529

**Comment by sumobi** [comment]
Replicated. You can set `0.91` for the main referral rate (in the AffiliateWP tab) which is a slight workaround and will generate a `0,91` referral fine. However if you have WooCommerce set to use Euros and `,` as a decimal separator then WooCommerce throws an error message if you try and use `0.91` as the referral rate when editing each variation.

**Comment by DrewAPicture** [comment]
Added a PR that fixes this: #2056 

Still need tests (if possible).

**Comment by DrewAPicture** [comment]
Added two tests.

**Comment by DrewAPicture** [comment]
And two more tests for the default decimal separator in rates.

**Comment by pippinsplugins** [comment]
Tests passed. PR merged.

**Comment by pippinsplugins** [comment]
RE-opening and reverting as this caused referrals to get generated as 0.00. See https://github.com/AffiliateWP/AffiliateWP/issues/2073

**Comment by pippinsplugins** [comment]
We'll need to add more robust unit tests for this.

**Comment by DrewAPicture** [comment]
OK, I just started another PR: #2231

**Comment by pippinsplugins** [comment]
@sumobi could you do some extensive testing on this one?

**Comment by sumobi** [comment]
@pippinsplugins @DrewAPicture @gspice 

Doesn't seem to be working correctly for me, perhaps I misunderstood the issue and PR?

Branch: `issue/1972-redux`

**AffiliateWP:**
Currency = Euro
Referral Rate Type = Percentage
Decimal Separator = comma `,`

**WooCommerce:**
Currency = Euro
Affiliate Rate on product = `0,5`

Product purchased = `€100.00`

Expected outcome = Referral amount of `€0,05`
Actual outcome = Referral amount of `€0,00`

Isn't the above what we're trying to fix? That is, if someone enters `0,5` for the referral rate on the WooCommerce product, we treat it at `0.5` (0.5%).

**Comment by pippinsplugins** [comment]
Is WooCommerce also setup with `,` as the decimal separator?

**Comment by sumobi** [comment]
@pippinsplugins No but I tried that and I still get a `€0,00` referral amount

**Comment by DrewAPicture** [comment]
@sumobi OK, could you please test this again and let me know if you see any change?

**Comment by DrewAPicture** [comment]
OK, there are now 10 separate tests in the PR:

```
5 = 5
-5 = 5
0.50 = 0.50
0,50 = 0.50
1.234,56 = 1234.56
1 234,56 = 1234.56
1 234.56 = 1234.56
1,234 = 1234
1.234 = 1234
1 234 = 1234
```

**Comment by sumobi** [comment]
@DrewAPicture First test worked, referral came through correctly. Going to hammer it some more, please hold.

**Comment by sumobi** [comment]
Can't seem to fault it! 👍

**Comment by DrewAPicture** [comment]
Merged #2231.
